orangecontrib/prototypes/widgets/owexplpred.py

```diff
@@ -0,0 +1,401 @@
+import sys
+import copy
+import logging
+import concurrent.futures
+from functools import partial
+
+from AnyQt.QtWidgets import (
+    QApplication, QFormLayout, QTableView,  QSplitter, QSizePolicy)
+from AnyQt.QtCore import Qt, QThread, pyqtSlot
+from PyQt5.QtGui import QSizePolicy
+import numpy as np
+import scipy.stats as st
+
+import Orange
+import Orange. evaluation
+from Orange.widgets.widget import OWWidget, Output, Input, Msg
+from Orange.widgets import gui, widget, settings
+from Orange.widgets.utils.itemmodels import TableModel
+from Orange.widgets.utils.sql import check_sql_input
+from Orange.base import Model
+from Orange.data import DiscreteVariable, ContinuousVariable, Domain, Table
+from Orange.widgets.utils.concurrent import (
+    ThreadExecutor, FutureWatcher, methodinvoke
+)
+
+
+class Task:
+
+    future = ...
+    watcher = ...
+    canceled = False
+
+    def cancel(self):
+        self.canceled = True
+        self.future.cancel()
+        concurrent.futures.wait([self.future])
+
+
+class ExplainPredictions(object):
+    """
+    Class used to explain individual predictions by determining the importance of attribute values.
+    All interactions between atributes are accounted for by calculating Shapely value.
+
+    Parameters
+    ----------
+    data : Orange.data.Table
+        table with dataset
+    model: Orange.base.Model
+        model to be used for prediction
+    error: float
+        desired error 
+    pError : float
+        p value of error
+    batchSize : int
+        size of batch to be used in making predictions, bigger batch size speeds up the calculations and improves estimations of variance
+    maxIter : int
+        maximum number of iterations per attribute
+    minIter : int
+        minimum number of iterations per attiribute
+    seed : int
+        seed for the numpy.random generator, default is 0
+
+    Returns:
+    -------
+    classValue: float
+        either index of predicted class or predicted value
+    table: Orange.data.Table
+        table containing atributes and corresponding contributions
+
+    """
+
+    def __init__(self, data, model, pError=0.05, error=0.05, batchSize=100, maxIter=59000, minIter=100, seed=0):
+        self.model = model
+        self.data = data
+        self.pError = pError
+        self.error = error
+        self.batchSize = batchSize
+        self.maxIter = maxIter
+        self.minIter = minIter
+        self.atr_names = DiscreteVariable(name='attributes',
```

Comment:
`atr_names` should be a StringVariable.

Another question is if the output table wouldn't be nicer transposed (column would be continuous attributes with names same as input columns).